### PR TITLE
Chunkinfo

### PR DESCRIPTION
**Pull request by jreadey**
Added arange chunk initiaiizer.

### CODE REVIEW

File: hsds/datanode_lib.py [hunk 1]
```diff
@@ -31,15 +31,15 @@
...
 from .util.dsetUtil import getChunkInitializer, getSliceQueryParam
-from .util.chunkUtil import getDatasetId, getChunkSelection
+from .util.chunkUtil import getDatasetId, getChunkSelection, getChunkIndex
 from .util.arrayUtil import arrayToBytes, bytesToArray, getShapeDims, jsonToArray
 from .util.hdf5dtype import createDataType, getItemSize
 
 from . import config
 from . import hsds_logger as log
 
 # supported initializer commands (just one at the moement)
```
Comment (mattjala) [inline hsds/datanode_lib.py:41]: No longer just one support command

**Review by mattjala — APPROVED** [review]
LGTM

File: hsds/datanode_lib.py [hunk 2]
```diff
@@ -829,6 +956,7 @@ async def get_chunk(
 
     # validate arguments
     if s3path:
+        """
```
Comment (mattjala) [inline hsds/datanode_lib.py:959]: Why remove this validation?

File: hsds/util/chunkUtil.py [hunk 3]
```diff
@@ -1170,3 +1170,35 @@ def chunkQuery(
         log.debug(f"   {i}: {rsp_arr[i]}")
 
     return rsp_arr
+
+def _find_min_pair(h5chunks, max_gap=None):
+    """ given a dict of chunk_map entries, return the two
+        chunks nearest to each other in the file.
```
Comment (mattjala) [inline hsds/util/chunkUtil.py:1176]: `return the two chunks` -> `return the indices of the two chunks`

File: hsds/chunk_dn.py [hunk 4]
```diff
@@ -366,8 +366,8 @@ async def GET_Chunk(request):
             num_bytes = s3size
         else:
             # list
-            num_bytes = np.prod(s3size)
-        log.debug(f"reading {num_bytes} from {s3path}")
+            num_bytes = np.sum(s3size)
```
[the remark this answers is not in the document]
Comment (jreadey) [reply hsds/chunk_dn.py:369]: no, it should be sum.  In this case s3size is a colon separated list of range lengths, so the sum is the total number of bytes to read.

File: hsds/datanode_lib.py [hunk 5]
```diff
@@ -24,18 +25,22 @@
...
-from .util.arrayUtil import arrayToBytes, bytesToArray
+from .util.dsetUtil import getChunkInitializer, getSliceQueryParam
+from .util.chunkUtil import getDatasetId, getChunkSelection, getChunkIndex
+from .util.arrayUtil import arrayToBytes, bytesToArray, getShapeDims, jsonToArray
 from .util.hdf5dtype import createDataType, getItemSize
 
 from . import config
 from . import hsds_logger as log
 
+# supported initializer commands (just one at the moement)
```
[the remark this answers is not in the document]
Comment (jreadey) [reply hsds/datanode_lib.py:42]: updated

File: hsds/datanode_lib.py (hunk 2 above)
Comment (jreadey) [reply hsds/datanode_lib.py:959]: Looks like this is not relevant after latest code changes.

File: hsds/util/chunkUtil.py [hunk 6]
```diff
@@ -698,6 +698,25 @@ def getChunkRelativePoint(chunkCoord, point):
...
+    Get the cannoncial size of the chunktable for a
+    given dataset and chunk shape"""
+    rank = len(shape_dims)
+    table_dims = []
+    for dim in range(rank):
+        dset_extent = shape_dims[dim]
+        chunk_extent = chunk_dims[dim]
+
+        if dset_extent > 0 and chunk_extent > 0:
+            table_extent = -(dset_extent // -chunk_extent)
```
[the remark this answers is not in the document]
Comment (jreadey) [reply hsds/util/chunkUtil.py:712]: No, this is little trick to compute the integer ceiling.
E.g.:   -(8 // -3)  -> 3
             8 // 3 -> 2

[the remark this answers is not in the document]
Comment (jreadey) [reply hsds/util/chunkUtil.py:712]: No, this is little trick to compute the integer ceiling.
E.g.:   -(8 // -3)  -> 3
             8 // 3 -> 2

[the remark this answers is not in the document]
Comment (mattjala) [reply hsds/util/chunkUtil.py:713]: Ah, I see. Is there a reason not to use `int(math.ceil(dset_extent // chunk_extent))`? Other parts of HSDS already import `math`.

**Review by mattjala — APPROVED** [review]

**Pull request merged**